File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/renderer/PhpClassRendererInterface.java
package com.github.tobi812.sprykerplugin.models.renderer;

import pav.sprykerFileCreator.model.renderer.dto.DocBlockItem;
import pav.sprykerFileCreator.model.renderer.dto.PhpClassInterface;

import java.util.List;

public interface PhpClassRendererInterface {

    String renderPhpClass(PhpClassInterface phpClass);

    String renderDocBlock(List<DocBlockItem> docBlockItems);

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/renderer/PhpClassRendererInterface.kt
package com.github.tobi812.sprykerplugin.models.renderer

import com.github.tobi812.sprykerplugin.models.renderer.dto.DocBlockItem
import com.github.tobi812.sprykerplugin.models.renderer.dto.PhpClassInterface
import com.jetbrains.php.lang.psi.elements.PhpClass

interface PhpClassRendererInterface {
    fun renderPhpClass(phpClass: PhpClassInterface): String
    fun renderDocBlock(docBlockItems: List<DocBlockItem>): String
    fun renderFactoryMethod(phpClass: PhpClass): String
}
